ipywidgets/static/widgets/js/manager.js

```diff
@@ -280,11 +280,26 @@ define([
         /**
          * Handle when a comm is opened.
          */
-        return this.create_model({
-            model_name: msg.content.data.model_name,
-            model_module: msg.content.data.model_module,
-            comm: comm,
-        }).catch(utils.reject("Couldn't create a model.", true));
+        // For < 4.x
+        var model;
+        if (msg.content.data.model_name) {
```

Comment:
I don't quite understand the whole process but that sounds like an awful lot of things to support. But yeah, if you want to do it, I won't say no :) I'd be happy if I had the good stack traces back, though.